src/sync/channel.zig

```diff
@@ -128,19 +143,27 @@ pub fn Channel(T: type) type {
 
         /// to deinit channels created with `create`
         pub fn destroy(channel: *Self) void {
+            const allocator = channel.allocator; // read allocator from channel before freeing it
             channel.deinit();
-            channel.allocator.destroy(channel);
+            allocator.destroy(channel);
         }
 
         pub fn close(channel: *Self) void {
             channel.closed.store(true, .monotonic);
-            channel.condition.broadcast();
         }
 
         pub fn send(channel: *Self, value: T) !void {
             if (channel.closed.load(.monotonic)) {
                 return error.ChannelClosed;
             }
+
+            const send_hook = channel.send_hook;
+            if (send_hook) |hook| {
+                if (!hook.before_send(hook, channel, value)) {
+                    return;
+                }
+            }
```

Comment:
i think adding an intermediate makes this much clearer

```suggestion
            if (send_hook) |hook| {
                const should_send_channel = hook.before_send(hook, channel, value)
                if (!should_send_channel) {
                    return;
                }
            }
```

Reply:
Im actually fine with removing it for now, then adding it back in #465 if needed.